lib/connectors/base/connector.rb

```diff
@@ -58,6 +62,10 @@ def initialize(configuration: {}, job_description: {})
 
       def yield_documents; end
 
+      def yield_with_handling_transient_errors(identifier: nil, &block)
```

Comment:
Should this PR introduce using this in the example connector too?

Reply:
Good call, I want to introduce a sample connector that uses it in a different PR.